example3/GroceryStoreExample/GroceryStore.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.Diagnostics.Metric;
+
+namespace GroceryStoreExample
+{
+    public class GroceryStore
+    {
+        private static Dictionary<string,double> price_list = new()
+        {
+            { "potato", 1.10 },
+            { "tomato", 3.00 },
+        };
+
+        private string store_name;
+
+        private Counter item_counter;
+        private Counter cash_counter;
+
+        public GroceryStore(string store_name)
+        {
+            this.store_name = store_name;
+
+            // TODO: Is GroceryStore a singleton? This example is only good guidance if it is.
+            // Otherwise we'd probably want MetricSource and the counters to be statics and
+            // Store should be a dimension on the counters rather than a static label.
+
+            MetricSource source = new MetricSource("GroceryStore", "1.0.0",
+                new Dictionary<string, string>()
+                {
+                    { "Store", store_name }
+                });
+
+
+            item_counter = new Counter("GroceryStore.item_counter",

```

Comment:
item_counter = source.CreateCounter("GroceryStore.item_counter",
                new string[] { "Item", "Customer" });
looks easy to write as the user is not requied to pass the source to the Counter's ctor.

The usage of new Ctor(), means the counter will have to return non-null all the time, right? Source.CreateCounter() allows us to return null if no listener expressed interest in this counter. (This is similar to ActivitySource.CreateActivity() as well)

Reply:
The OpenTelemetry.Metric.API project allows new DotnetMeter(name,ver).CreateCounter(countername) though.

Reply:
> The usage of new Ctor(), means the counter will have to return non-null all the time, right?

Yep! I doubt we'd want to return null even if we had the option to do so with a factory. Even if no listener is interested right away, they might be interested later. And if we don't create the counter now then there will no way to send data later.